File: CodeForces/Div.4/#886/B/main.py
def main():
    t = int(input())

    for i in range(t):
        n = int(input())

        res = -1
        win = 0
        for i in range(n):
            a, b = map(int, input().split())

            if a <= 10 and b > res:
                win = i + 1
                res = b
        
        print(win)
main()

File: CodeForces/Div.4/#886/B/main.cpp
#include <bits/stdc++.h>

using namespace std;

#define endl '\n'
typedef long long i64;
typedef int i32;
typedef short i16;
typedef vector<i64> vi64;
typedef vector<vi64> vv;
typedef string str;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  i64 t;
  cin >> t;

  while (t--) {
    i64 n;
    cin >> n;

    i64 res = -1, win = 0;
    for (i64 i = 1; i <= n; i++) {
      i64 rr, qq;
      cin >> rr >> qq;

      if (qq > res and rr <= 10) {
        win = i;
        res = qq;
      }
    }

    cout << win << endl;
  }

  return 0;
}
